jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/thread/QueuedThreadPool.java

```diff
@@ -764,7 +762,10 @@ public int getMaxAvailableThreads()
     @ManagedAttribute("utilization rate of threads executing unleased jobs")
     public double getUtilizationRate()
     {
-        return (double)getUtilizedThreads() / (getMaxThreads() - getLeasedThreads());
+        int available = getMaxThreads() - getLeasedThreads();
+        if (available <= 0)
+            throw new IllegalStateException("too many leased threads");
+        return (double)getUtilizedThreads() / available;
```

Comment:
Unnecessary, cannot be that `maxThread==leasedThreads` since `ThreadPoolBudget` avoids that.
I'd restore the old code.